tsdownsample/downsampling_interface.py

```diff
@@ -1,54 +1,102 @@
+"""AbstractDownsampler interface-class, subclassed by concrete downsamplers."""
+
+__author__ = "Jeroen Van Der Donckt"
+
+import re
 import warnings
 from abc import ABC, abstractmethod
 from types import ModuleType
-from typing import Callable
+from typing import Callable, List, Optional, Tuple, Union
 
 import numpy as np
-import pandas as pd
 
 
-class DownsampleInterface(ABC):
-    def __init__(self, name: str) -> None:
+class AbstractDownsampler(ABC):
+    """AbstractDownsampler interface-class, subclassed by concrete downsamplers."""
+
+    def __init__(self, name: str, dtype_regex_list: Optional[List[str]] = None):
         self.name = name
+        self.dtype_regex_list = dtype_regex_list
```

Comment:
The `name` property is only used for formatting the "repr" and error messages. I think this can be achieved as well with using the `self.__class__.__name__` attribute.

![image](https://user-images.githubusercontent.com/38005924/206840819-061ee323-a66a-4c57-be02-67ea129cad7e.png)

Reply:
Oops, was a left-over from an intitial singleton implementation..